tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -79,19 +95,65 @@ public void Run(JsonDrivenTestCase testCase)
         }
 
         // private methods
-        private void Run(BsonDocument shared, BsonDocument test)
+        private void CheckClusterTopology(BsonArray allowedTopologies)
+        {
+            var allowedClusterTypes = allowedTopologies.Select(MapToplogyToClusterType);
+            RequireServer.Check().ClusterTypes(allowedClusterTypes.ToArray());
```

Comment:
Minor style nit: I prefer the call to ToArray on the first line. I think we should never have lazy enumerables unless it is absolutely on purpose.

            var allowedClusterTypes = allowedTopologies.Select(MapToplogyToClusterType).ToArray();
            RequireServer.Check().ClusterTypes(allowedClusterTypes);

Putting ToArray on the first line also changes the type of allowedClusterTypes to an array which is much easier to inspect when debugging.